### ISSUE TITLE
Can't upload the dataset file with the same dataset name

### ISSUE THREAD

**Issue by YunliangHuang** [hning86/azuremlps#37]
Hi,

I tried to upload a dataset file with the same dataset name, but always hit the below error. There should be a parameter like -Overwrite or -Force to allow users to upload a content updated dataset file with the same dataset name. 

Upload-AmlDataset : Error: [409 (Duplicate found for asset name: workspace: 0c84945885e447a997389661bf0077d9)]: {"Error":{"Code":"UserError","Message":"Duplicate found for asset name: 
workspace: 0c84945885e447a997389661bf0077d9","Target":null,"Details":[],"InnerError":null,"DebugInfo":null}}
At line:1 char:1
+ Upload-AmlDataset -FileFormat GenericTSVNoHeader -UploadFileName 'G:\ ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Upload-AmlDataset], AmlRestApiException
    + FullyQualifiedErrorId : AzureML.Contract.AmlRestApiException,AzureML.PowerShell.UploadDataset
 
Thanks,
Tim

**Comment by quistador** [comment]
Yes, there should be an -Overwrite or -Force parameter.  I've spent a few hours trying to understand the underlying APIs but I've been unsuccessful.  I can't find any publicly available documentation, making it very hard to contribute a fix here. 


I did find a workaround that, depending on your situation, might work for you:  instead of calling 'Upload-AmlDataset' to replace a dataset, use it to upload a new dataset, then use 'Replace-AmlExperimentUserAsset' so that your experiment consumes the new dataset.  Something like:




`$datasetToReplace = Get-AmlDataset -Scope Experiment -ExperimentId | where (_condition_)
$newDataSet =  $amlDataSet | where (_condition_)
$experiment.ExperimentIdReplace-AmlExperimentUserAsset -ExperimentId $experiment.ExperimentId -AssetType Dataset -ExistingAsset $datasetToReplace -NewAsset $newDataSet`
